Sources/Weather/OAWeatherHelper.mm

```diff
@@ -570,10 +608,11 @@ - (void)onProgressUpdate:(OAWorldRegion *)region success:(BOOL)success
         [mapViewController.mapLayers.weatherLayerHigh updateWeatherLayer];
         [mapViewController.mapLayers.weatherContourLayer updateWeatherLayer];
 
-        [self calculateCacheSize:region onComplete:^()
-        {
-            dispatch_group_leave(_forecastGroupDownloader);
-        }];
+        dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
So you use the mapViewController not on the main thread but calculate the cache size on the main one. Looks highly suspicious and confusing.